src/sinks/websocket/sink.rs

```diff
@@ -301,7 +307,10 @@ impl WebSocketSink {
                         Ok(()) => {
                             finalizers.update_status(EventStatus::Delivered);
 
-                            let message = Message::text(String::from_utf8_lossy(&bytes));
+                            let message = match self.encode_as_binary {
+                                true => Message::binary(bytes),
+                                false => Message::text(String::from_utf8_lossy(&bytes)),
+                            };
```

Comment:
I personally find the match on a binary a little obtuse, so generally perfer:

```suggestion
                            let message = if encode_as_binary {
                                Message::binary(bytes)
                            } else {
                                Message::text(String::from_utf8_lossy(&bytes))
                            };
```

Reply:
updated